src/otx/core/metrics/fmeasure.py

```diff
@@ -780,8 +783,38 @@ def classes(self) -> list[str]:
         return self.label_info.label_names
 
 
+class MeanAveragePrecisionFMeasure(MetricCollection):
+    """Computes the mean AP with f-measure for a resultset."""
```

Comment:
```suggestion
    """Computes the mean AP with f-measure for a resultset.
    
    NOTE: IMPORTANT!!! Do not use this metric to evaluate a F1 score on a test set.
        This is because it can pollute test evaluation.
        It will optimize the confidence threshold on the test set by doing line search on confidence threshold axis.
        The correct way to obtain the test set F1 score is to use the best confidence threshold obtained from the validation set.
        You should use `--metric otx.core.metrics.fmeasure.FMeasureCallable` override to correctly obtain F1 score from a test set.
    """
```